lib/IR/IR.cpp

```diff
@@ -573,6 +580,27 @@ void IRFunction::dump(llvm::raw_ostream &OS) const {
   OS.flush();
 }
 
+std::string IRFunction::toString() const {
+  std::string storage;
+  llvm::raw_string_ostream os(storage);
+  dump(os);
+  return os.str();
+}
+
+// Dump a textual representation to default stream
+void Instruction::dump() const { dump(llvm::outs()); }
+
+/// Dump to string.
+std::string Instruction::toString() const {
+  std::string storage;
+  llvm::raw_string_ostream os(storage);
+  dump(os);
+  return os.str();
+}
+
+/// Dump a textual representation of the function into provided output stream.
+void Instruction::dump(llvm::raw_ostream &OS) const { dumpOperands(OS); }
```

Comment:
I think it would be useful to dump out the instruction name and type/kind as well.

Reply:
Instructions are Values. And Values have a dump method already. No need to redefine it.